ddtrace/_trace/span.py

```diff
@@ -541,6 +541,51 @@ def set_exc_info(
 
         core.dispatch("span.exception", (self, exc_type, exc_val, exc_tb))
 
+    def record_exception(
+        self,
+        exception: Exception,
```

Comment:
nit: Can we use `BaseException` here? This aligns the type with otel: https://github.com/open-telemetry/opentelemetry-python/blob/a7fe4f8bac7fa36291c6acf86982bbb356e3ae6d/opentelemetry-api/src/opentelemetry/trace/span.py#L172

```suggestion
        exception: BaseException,
```